### ISSUE TITLE
[clang-tidy] bugprone-use-after-move wrong warning within implementation of move constructor

### ISSUE THREAD

**Issue by SunBlack** [llvm/llvm-project#80693]
Given following code:

```cpp
#include <utility>

class External{};
class External2{};

class A
{
public:
  A()
    : MyObj(new External) {
  }

  A(A&& other)
  {
    std::swap(MyObj, other.MyObj);
  };
  
  ~A() {
    delete MyObj;
  };
protected:
  External* MyObj;
};


class B : public A
{
public:
  B()
    : MyObj2(new External2) {
  }

  B(B&& other)
    : A(std::move(other))
  {
    std::swap(MyObj2, other.MyObj2);
  };
  
  ~B() {
    delete MyObj2;
  };
protected:
  External2* MyObj2;
};
```

This code causes:
```
test.cpp:36:23: warning: 'other' used after it was moved [bugprone-use-after-move]
   36 |     std::swap(MyObj2, other.MyObj2);
      |                       ^
test.cpp:34:7: note: move occurred here
   34 |     : A(std::move(other))
      |       ^
```

I think this is a false-positve, as `MyObj2` should be still valid after the parent class was moved as the parent class cannot move this.

**Comment by HerrCai0907** [comment]
Yes, that is false positive, but it's still a danger behavior if you change the implement in `A(A&& other)`.
I think we should avoid to do cross-function analysis in clang-tidy check.
